tests/vmi_networking_test.go

```diff
@@ -568,6 +570,60 @@ var _ = Describe("Networking", func() {
 			checkLearningState(learningDisabledVMI, "0")
 		})
 	})
+
+	Context("VirtualMachineInstance with dhcp options", func() {
+		BeforeEach(func() {
+			tests.BeforeTestCleanup()
+		})
+
+		FIt("should offer extra dhcp options to pod iface", func() {
+			userData := "#cloud-config\npassword: fedora\nchpasswd: { expire: False }\n"
+			dhcpVMI := tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskFedora), userData)
+			tests.AddExplicitPodNetworkInterface(dhcpVMI)
+
+			dhcpVMI.Spec.Domain.Resources.Requests[k8sv1.ResourceName("memory")] = resource.MustParse("1024M")
+			dhcpVMI.Spec.Domain.Devices.Interfaces[0].DHCPOptions = &v1.DHCPOptions{
+				BootFileName:   "config",
+				TFTPServerName: "tftp.kubevirt.io",
+			}
+
+			_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(dhcpVMI)
+			Expect(err).ToNot(HaveOccurred())
+
+			tests.WaitUntilVMIReady(dhcpVMI, tests.LoggedInFedoraExpecter)
+
+			err = tests.CheckForTextExpecter(dhcpVMI, []expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: "#"},
+				&expect.BSnd{S: "sudo dhclient -1 -r -d eth0\n"},
+				&expect.BExp{R: "#"},
+				&expect.BSnd{S: "echo $?\n"},
+				&expect.BExp{R: "0"},
+			}, 15)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = tests.CheckForTextExpecter(dhcpVMI, []expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: "#"},
+				&expect.BSnd{S: "sudo dhclient -1 -sf /usr/bin/env --request-options subnet-mask,broadcast-address,time-offset,routers,domain-search,domain-name,domain-name-servers,host-name,nis-domain,nis-servers,ntp-servers,interface-mtu,tftp-server-name,bootfile-name eth0\n"},
+				&expect.BExp{R: "new_tftp_server_name=tftp.kubevirt.io"},
+				// &expect.BSnd{S: "echo $?\n"},
+				// &expect.BExp{R: "0"},
+			}, 15)
+
+			// Was planning on executing /usr/bin/env and checking for each option passed in
```

Comment:
Yes, dhclient only passes the environmentvariables to the binary in `-sf`, so it is not persisted in your session. You can try to do two `Bexp` in a row:

```go
&expect.BExp{R: "new_tftp_server_name=tftp.kubevirt.io"},
&expect.BExp{R: "new_bootfile_name=config"}
```

but I don't know if the order is always the same.

Another option is to write the env output somewhere (e.g. `/dhcp-env`):

```go
&expect.BSnd{S: "sudo dhclient -1 -sf /usr/bin/env --request-options subnet-mask,broadcast-address,time-offset,routers,domain-search,domain-name,domain-name-servers,host-name,nis-domain,nis-servers,ntp-servers,interface-mtu,tftp-server-name,bootfile-name eth0 > /dhcp-env\n"},
```

and then cat and check:

```go
&expect.BSnd{S: "cat /dhcp-env\n"},
&expect.BExp{R: "new_tftp_server_name=tftp.kubevirt.io"},
&expect.BSnd{S: "cat /dhcp-env\n"},
&expect.BExp{R: "new_bootfile_name=config"}
```

Reply:
Have to tee it but that works like a charm actually :smile: !